libs/api/ebpf_api.cpp

```diff
@@ -2323,22 +2322,32 @@ _initialize_ebpf_object_from_file(
     EBPF_LOG_ENTRY();
     ebpf_result_t result = EBPF_SUCCESS;
 
-    new_object->file_name = cxplat_duplicate_string(path);
+    if (std::holds_alternative<std::string>(file_or_data)) {
+        std::string path = std::get<std::string>(file_or_data);
+        new_object->file_name = cxplat_duplicate_string(path.c_str());
+    } else {
+        new_object->file_name = cxplat_duplicate_string("memory");
```

Comment:
nit: maybe make this more unique? eg. EBPF_PROG_ID_MEMORY_BUF or similar

Reply:
Looks like libbpf uses the name from the opts file, so I will switch it.